src/format/multifactor/AuthenticationFactorInfo.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2024 KeePassXC Team <team@keepassxc.org>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 or (at your option)
+ * version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef KEEPASSXC_AUTHENTICATIONFACTORINFO_H
+#define KEEPASSXC_AUTHENTICATIONFACTORINFO_H
+
+#include "format/multifactor/AuthenticationFactorGroup.h"
+#include <QCoreApplication>
+#include <QSharedPointer>
+
+class AuthenticationFactorGroup;
+
+class AuthenticationFactorInfo : public QObject
+{
+    Q_OBJECT
+
+public:
+    explicit AuthenticationFactorInfo() = default;
+    ~AuthenticationFactorInfo() override = default;
```

Comment:
There is no reason to inherit QObject if you aren't going to take advantage of the class. Advantages include: using signal/slots, parent-child dependency and cleanup, multi-threaded data sharing.

This applies to other classes in this PR.